dist/mapdc.js

```diff
@@ -48641,6 +48646,31 @@ function rasterChart(parent, useMap, chartGroup, _mapboxgl) {
     return _legend;
   };
 
+  _chart.deleteLayerLegend = function (currentLayer, deleteLayer, prevLayerId) {
+    var legend = document.getElementsByClassName('legend');
+    if (legend.length && currentLayer !== "master") {
+      // the case that has 3 or more layers will be handled in getLegendStateFromChart function in stacked-legend.js by removing extra legend
+      if (prevLayerId && prevLayerId === "master") {
+        legend[deleteLayer].remove();
+      } else if (currentLayer !== deleteLayer && prevLayerId < deleteLayer) {
+        return;
+      } else if (legend[deleteLayer]) {
+        legend[deleteLayer].remove();
+      } else if (deleteLayer && deleteLayer !== 'master' && legend.length === 1 && legend[0]) {
+        legend[0].remove();
+      }
+    }
+  };
+
+  _chart.destroyChartLegend = function () {
+    var legend = document.getElementsByClassName('legend');
```

Comment:
Is it possible that this could be simplified as:
```
_chart.selectAll('legend').remove()
```
D3 will remove them only if they exist, and will remove them all.

Reply:
@biovisualize, that doesn't remove the element from the DOM. I tried and didn't work out.